tools/ClassGen/NodeGen.cpp

```diff
@@ -592,6 +592,18 @@ int main(int argc, char **argv) {
                     "Produces the updated weight that needs to be used "
                     "instead of Weight for the next iteration.");
 
+  //===--------------------------------------------------------------------===//
+  //             Nodes used for debugging/profiling/printing
+  //===--------------------------------------------------------------------===//
+
+  BB.newNode("TraceEvent")
+      .addInput("Data")
+      .addMember(MemberType::String, "EventName")
+      .addMember(MemberType::String, "EventType")
```

Comment:
I don't really because we'd need to keep the enum in line with the TraceEvent spec but we probably wont check it internally. A string lets us pass it through directly without inspection. I also don't feel strongly about this so willing to be convinced.